src/interfaces/tests/ExclusiveProductDataController.spec.js

```
@@ -0,0 +1,48 @@
+const ExclusiveProductDataController = require('../ExclusiveProductDataController');
+const sampleData = require('../../../TestData');
+const ExclusiveProductStringCreator = require('../../application/use_cases/ExclusiveProductStringCreator');
+
+const TransferController = require('../MessagingClass');
+
+const messageService = new TransferController();
+
+describe('dataControl', () => {
+  it('Should call getExclusiveProductString', () => {
+    ExclusiveProductStringCreator.getExclusiveProductString = jest.fn();
+    ExclusiveProductDataController.dataControl(sampleData.exclusiveProductsAll);
+    expect(ExclusiveProductStringCreator.getExclusiveProductString).toHaveBeenCalledTimes(1);
+  });
+  it('Should call sendMessage', () => {
+    TransferController.prototype.sendMessage = jest.fn();
+    ExclusiveProductDataController.dataControl(sampleData.exclusiveProductsAll);
+    expect(messageService.sendMessage).toHaveBeenCalledTimes(1);
+  });
+  it('Should error if addSachets is incorrect ', () => {
```

Should throw an error - just a matter of taste.